### PR TITLE
Identify inner build in consuming project

### PR DESCRIPTION
**Pull request by rainersigwald**
This is a proposed ProjectReference protocol change and the required new target (+ dependency on NuGet.Frameworks).

This allows solving the double-evaluation-of-ProjectReferences problem by ensuring that a) every referencing project calls into the same cached `GetTargetFrameworks`-with-no-TF-set evaluation of the reference, and b) single-targeted projects will use the same evaluation for their build.

It is also cacheable, but that's not done here.

Builds on #2458, since that simplified the ProjectReference flow.

### CODE REVIEW

File: src/Tasks/Microsoft.Common.CurrentVersion.targets [hunk 1]
```diff
@@ -1555,6 +1541,60 @@ Copyright (C) Microsoft Corporation. All rights reserved.
...
+    Builds the GetTargetFrameworkProperties target of all existent project references,
+    passing $(TargetFrameworkMoniker) as $(ReferringTargetFramework) and sets the
+    SetTargetFramework metadata of the project reference to the value that is returned.
+
+    This allows a cross-targeting project to select how it should be configured to
+    build against the most appropriate target for the referring target framework.
+
+    ======================================================================================
+  -->
+  <Target Name="_GetProjectReferenceTargetFrameworkProperties">
```
Comment (rainersigwald) [inline src/Tasks/Microsoft.Common.CurrentVersion.targets]: This is no longer needed if the above multitargeting handling is comprehensive. If we go with the try-the-new-way-and-don't-fail-if-it-doesn't-work approach, it'd serve as the compat fallback.

**Comment by rainersigwald** [comment]
Left to do:

- [ ] Figure out if the breaking change is ok (or implement `<MSBuild Dont
- [ ] Figure out TF-compat-oracle story
- [ ] Productize error messages
- [ ] Compat testing on cli
- [ ] Perf

**Comment by cdmihai** [comment]
@dotnet-bot test this please

**Comment by AndyGerlicher** [comment]
Closing in favor of #2595

**Pull request closed without merging**